packages/backend/contracts/AttesterDescription.sol

```diff
@@ -0,0 +1,46 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8;
+
+import {Unirep} from '@unirep/contracts/Unirep.sol';
+
+contract AttesterDescription {
+  bool private isValid;
+  bytes4 internal constant MAGICVALUE = 0x1626ba7e;
+
+  function supportsInterface(bytes4 interfaceId) external pure returns (bool) {
+    return interfaceId == 0x93c93c46;
+  }
+
+  function isValidSignature(
+    bytes32 _hash,
+    bytes memory _signature
+  ) public view returns (bool) {
+    (bytes32 r, bytes32 s, uint8 v) = splitSignature(_signature);
+    return ecrecover(getEthSignedMessageHash(_hash), v, r, s) == msg.sender;
+  }
+
+  function getEthSignedMessageHash(
+    bytes32 _messageHash
+  ) public pure returns (bytes32) {
+    return
+      keccak256(
+        abi.encodePacked('\x19Ethereum Signed Message:\n32', _messageHash)
+      );
+  }
+
+  function splitSignature(
+    bytes memory sig
+  ) public pure returns (bytes32 r, bytes32 s, uint8 v) {
+    require(sig.length == 65, 'invalid signature length');
+
+    assembly {
+      r := mload(add(sig, 32))
+      s := mload(add(sig, 64))
+      v := byte(0, mload(add(sig, 96)))
+    }
+  }
+
+  function getInterfaceId() external pure returns (bytes4) {
+    return this.isValidSignature.selector;
+  }
```

Comment:
where is it used?

Reply:
it's used to calculate the constant '0x93c93c46' in
```js
    let supportsInterface = false
    try {
      supportsInterface = await contract.supportsInterface('0x93c93c46')
    } catch (_) {
      // assume the function call fails and the interface is not supported
      res.status(401)
      passed = false
    }
```

Reply:
Can it be like
```ts
    let supportsInterface = false
    try {
      supportsInterface = await contract.supportsInterface(contract.getInterfaceId())
    } catch (_) {
      // assume the function call fails and the interface is not supported
      res.status(401)
      passed = false
    }
```

or it can change the `supportsInterface` function
```solidity
  function supportsInterface(bytes4 interfaceId) external pure returns (bool) {
    return interfaceId == getInterfaceId();
  }
```

Reply:
in that case, even if the contract doesn't implement 
```js 
function isValidSignature(
    bytes32 _hash,
    bytes memory _signature
  ) public view returns (bool)
  ```
  it will still pass, so `interfaceId` needs to be a constant